compiler/rustc_typeck/src/expr_use_visitor.rs

```diff
@@ -27,14 +27,38 @@ use rustc_span::Span;
 /// employing the ExprUseVisitor.
 pub trait Delegate<'tcx> {
     // The value found at `place` is either copied or moved, depending
-    // on mode.
-    fn consume(&mut self, place_with_id: &PlaceWithHirId<'tcx>, mode: ConsumeMode);
+    // on mode. Where `diag_expr_id` is the id used for diagnostics for `place`.
+    //
+    // The reson for a separate expr_id for diagonostics is to support cases
```

Comment:
s/reson/reason/ (I can't seem to find the "automated suggestion" UI in github anymore...)

Reply:
![image](https://user-images.githubusercontent.com/4193035/98084596-0bd73b00-1e4a-11eb-8c4d-117de2036f4a.png)